src/cms.js

```diff
@@ -26,7 +27,7 @@ class CMS {
    */
   constructor (keychain) {
     if (!keychain) {
-      throw new Error('keychain is required')
+      throw errcode(new Error('keychain is required'), 'ERR_KEYCHAIN_REQUIRED')
```

Comment:
`errcode` already creates a new `Error` if we provide a string. So, I think passing only a string is cleaner and uniform with other modules for `js-ipfs` and `js-libp2p`.

```suggestion
      throw errcode('keychain is required', 'ERR_KEYCHAIN_REQUIRED')
```

Reply:
If we do that, then the stack trace starts inside the `err-code` module which is confusing as the first line of the stack trace is not where the error occurred.

Fixing that is the whole point of this PR.

Reply:
That is a good point! We have that anti-pattern in a big part of the `js-ipfs` and `js-libp2p` codebase. And in that way, we should change that as well! I

Reply:
Yeah, we should update those too..